### ISSUE TITLE
Enable Server side rendering (SSR)

### ISSUE THREAD

**Issue by davidyuk** [aeternity/mdw-frontend#136]
was disabled in https://github.com/aeternity/mdw-frontend/pull/84
also doing
```bash
$ curl https://explorer.aeternity.io/account/ak_jupBUgZNbcC4krDLR3tAkw1iBZoBbkNeShAq4atBtpFWmz36r
```
I don't see actual page markup as it should be with SSR enabled

- to have faster first rendering time
- to have rich preview in social media
- to allow indexing by search engines

why are using Nuxt after all?

**Comment by marc0olo** [comment]
> why are using Nuxt after all?

I heard that question now multiple times. actually we currently just try to display as much info as possible which was missing before.

so far we didn't find the time to improve these things. feel free to provide a PR if you wanna fix this ;-)

**Comment by davidyuk** [comment]
This is mostly a deployment issue (cc: @mmpetarpeshev), from the code side we need just to revert the mentioned PR

**Comment by mmpetarpeshev** [comment]
@davidyuk Not sure that I understand the issue here. What is the deployment issue ?

**Comment by davidyuk** [comment]
Can we setup the deployment of mdw-frontend in SSR mode? We need it firstly for staging because some stuff may be broken now.

**Comment by marc0olo** [comment]
@venimus what was the reason for disabling it in #84?

**Comment by venimus** [comment]
It is hosted as a static site

**Comment by marc0olo** [comment]
I need to recall this topic. it's still confusing for me.

@davidyuk does SSR make sense for this application? in what form did we use it for the things you mentioned above? from my perspective I am not sure if we need indexing of search engines. what could be cool of course is the rich preview in social media. not sure if the faster rendering time is that important for this application.

@venimus what was the reason for disabling it?

general question:
- if we decide to drop SSR, should we also get rid of nuxt?

**Comment by davidyuk** [comment]
> does SSR make sense for this application?

I think it is worth maintaining it. Also for educational purposes, to estimate is it useful to have SSR in other projects, to have devs that know how it works in Nuxt.

> in what form did we use it for the things you mentioned above?

I don't know. It was implemented explicitly in https://github.com/aeternity/aepp-blockchain-explorer/issues/490 and related tickets, but I can't find any reasoning, probably it somewhere in old chats.

> if we need indexing of search engines

I've tried to google my Ethereum address, and I got its details on etherscan.io, googling my ae address I got only aeknow.org. Ability to google it looks useful for users not familiar with the concept of blockchain explorer.

> if we decide to drop SSR, should we also get rid of nuxt?

would be better to use the same tool in different projects (vue cli), but it is not necessary, moreover, we might decide to enable SSR later

**Comment by venimus** [comment]
> @venimus what was the reason for disabling it?

because it does not really make sense for the app, all the content is dynamically rendered. However the explorer is/was hosted as static site, there is no "server side rendering". In addition the SSR option in the config in fact simply prerenders static versions of all the urls, which results in prebuilding a snapshot of all transactions, which is insane in terms of building time and disk space. Until things change conceptually for the app it simply cant use SSR.

**Comment by venimus** [comment]
Yes, I get what you mean, I just want to elaborate that the option i turned off is not related to what we are discussing. I’ll check with Petar to see how we can host it so SSR works and what would be needed to be changed

**Comment by venimus** [comment]
Sure, it’s just in a discussion state still
